example/src/main/java/com/stripe/example/activity/USBankAccountPaymentMethodActivity.kt

```diff
@@ -0,0 +1,362 @@
+                        "Examples to debit the bank account specified above for any amount owed " +
+                        "for charges arising from your use of Non-Card Payment Examples’ " +
+                        "services and/or purchase of products from Non-Card Payment Examples, " +
+                        "pursuant to Non-Card Payment Examples’ website and terms, until this " +
+                        "authorization is revoked. You may amend or cancel this authorization at " +
+                        "any time by providing notice to Non-Card Payment Examples with 30 " +
+                        "(thirty) days notice." +
+
+                        "If you use Non-Card Payment Examples’ services or purchase additional" +
+                        "products periodically pursuant to Non-Card Payment Examples’ terms, you" +
+                        "authorize Non-Card Payment Examples to debit your bank account " +
+                        "periodically. Payments that fall outside of the regular debits " +
+                        "authorized above will only be debited after your authorization is " +
+                        "obtained.",
+                    color = Color.Gray,
+                    fontSize = 8.sp
+                )
+            }
+        }
+    }
+
+    private fun createAndConfirmSetupIntent(
+        params: PaymentMethodCreateParams
+    ) {
+        viewModel.createSetupIntent(
+            country = "us"
+        ).observe(
+            this
+        ) {
+            it.onSuccess { responseData ->
+                viewModel.status.postValue(
+                    viewModel.status.value +
+                        "\n\nStarting SetupIntent confirmation"
+                )
+                lifecycleScope.launch {
+                    val result = runCatching {
+                        stripe.confirmSetupIntent(
+                            ConfirmSetupIntentParams.create(
+                                paymentMethodCreateParams = params,
+                                clientSecret = responseData.getString("secret")
+                            )
+                        )
+                    }
+                    result.fold(
+                        onSuccess = { setupIntent ->
+                            if (setupIntent.requiresAction() &&
+                                setupIntent.nextActionType ==
+                                StripeIntent.NextActionType.VerifyWithMicrodeposits
+                            ) {
+                                postResults(PaymentResult.Completed)
+                            } else {
+                                postResults(PaymentResult.Canceled)
+                            }
+                        },
+                        onFailure = {
+                            postResults(PaymentResult.Failed(it))
+                        }
+                    )
+                }
+            }
```

Comment:
Ideally all this logic should live on the viewmodel or the `confirmSetupIntent` execution will be cancelled on configuration changes. Also this will serve as guidelines for people integrating this functionality, and this could mislead developers to run API calls in the view layer right?